ansible/roles/xsnippet_api/tasks/main.yml

```diff
@@ -0,0 +1,102 @@
+---
+
+- name: Update apt cache if needed
+  ansible.builtin.apt:
+    update_cache: true
+    cache_valid_time: 3600
+  become: true
+
+- name: Install prerequisite packages
+  ansible.builtin.apt:
+    name:
+      - libpq-dev
+      - python3-alembic
+      - python3-psycopg2
+    state: present
+  become: true
+
+- name: Create '{{ xsnippet_api_user }}' user
+  ansible.builtin.user:
+    name: "{{ xsnippet_api_user }}"
+    shell: /sbin/nologin
+    home: /var/lib/{{ xsnippet_api_user }}
+  become: true
+
+- name: Create target directory for sources
+  ansible.builtin.file:
+    path: "{{ xsnippet_api_src_root }}"
+    mode: u=rwx,g=rx,o=rx
+    state: directory
+  become: true
+
+- name: Install the sources artifact
+  ansible.builtin.unarchive:
+    src: "{{ xsnippet_api_src_artifact }}"
+    dest: "{{ xsnippet_api_src_root }}"
+    extra_opts:
+      - --strip-components=1
+    owner: root
+    group: root
+    mode: u=rwx,g=rx,o=rx
+    remote_src: true
+  register: xsnippet_api_src_unarchive
+  become: true
+
+- name: Run database migrations
+  ansible.builtin.command:
+    cmd: python3 -m alembic.config upgrade head
+    chdir: "{{ xsnippet_api_src_root }}"
+  environment:
+    ROCKET_DATABASE_URL: "{{ xsnippet_api_database_url }}"
+  when: xsnippet_api_src_unarchive.changed
```

Comment:
@malor Do we actually want this? With this in-place, there are no *changed* tasks on second rerun. On the other hand we have to run migrations if database url is changed as well.

Reply:
Hm.. we can do that, I think. I initially planned to download sources to a temporary directory since we don't need them later on. I discarded that idea at some point though.

Reply:
I ended up using systemd's `ExecStartPre=` that executes a binary before starting the main application, and I've created `/opt/xsnippet-api/database-upgrade` shell script that invokes `alembic`.